plotter/lineplotter.cpp

```diff
@@ -408,35 +408,25 @@ void LinePlotter::testThreads(QCPRange range) {
     if(ui->plot->graphCount() == 0) {
         return;
     }
-
     int graphIndex = 0;
+
     for(int i=0; i<arrays.size(); i++) {
         int xDim = guess_best_xdim(arrays[i]);
         QCPGraph *graph = ui->plot->graph(graphIndex);
 
+        if(arrays[i].dataExtent().size() == 1) {
+            graphIndex += 1;
+        } else if(arrays[i].dataExtent().size() == 2) {
+            graphIndex += arrays[i].dataExtent()[2-xDim];
+        }
         if(graph->dataCount() == 0) {
```

Comment:
As far as i know this shouldn't lead to seq faults. The access with dataExtent().size() was fine in my tests and it's still there.

Reply:
it is not the call itself but the concurrent access, methinks